napalm_ios/ios.py

```diff
@@ -2128,3 +2132,7 @@ def get_config(self, retrieve='all'):
             configs['running'] = output
 
         return configs
+
+    @property
+    def file_system(self):
+        return self.dest_file_system or self._discover_file_system()
```

Comment:
Can't we just do:

In __init__, line 86:
```
# Note extra underscore.
self._dest_file_system = optional_args.get('dest_file_system', None)
```

```
@property
def dest_file_system(self):
    if self._dest_file_system is None:
        self._dest_file_system = self._discover_file_system()
    return self._dest_file_system
```

Then anytime object.dest_file_system gets accessed it will use the current value (if there is one) or will call the _discover_file_system() method.

Then we don't have to do the artificial calls in `replace` and `merge`.